### ISSUE TITLE
Incorrect handling of special characters in strings

### ISSUE THREAD

**Issue by dtolnay** [saghm/json-color#1]
```rust
extern crate json_color;

use json_color::Colorizer;

fn main() {
    let colorizer = Colorizer::arbitrary();

    println!("{}", colorizer.colorize_json_str("\"a\\nb\"").unwrap());
}
```

Expected:

```
"a\nb"
```

Actual:

```
"a
b"
```

Since you are using serde_json anyway, consider implementing json-color as a [`Formatter`](https://docs.serde.rs/serde_json/ser/trait.Formatter.html) to take advantage of our logic for rendering escape sequences, among other things. Something like:

```rust
pub fn colorize_json_str(&self, s: &str) -> Result<String, Error> {
    let mut wr = Vec::new();
    let formatter = ColorFormatter { colorizer: self, indent_level: 0 };
    {
        let mut de = serde_json::Deserializer::from_str(s);
        let mut ser = serde_json::Serializer::with_formatter(&mut wr, formatter);
        serde_transcode::transcode(&mut de, &mut ser)?;
    }
    Ok(String::from_utf8(wr).unwrap())
}
```

A good starting point would be [`rq`'s colorizing formatter implementation](https://github.com/dflemstr/rq/blob/v0.9.2/src/value/json.rs#L132). Note that every method has a sensible default implementation so you would only need to override the ones that need to be colored.

@saghm

**Comment by saghm** [comment]
Thanks for the suggestion! Back when I wrote this, I didn't really expect anyone else to use it, so I implemented it in the simplest way possible. (I also haven't played around with serde nearly as much as I'd like to have, so I was completely unaware that there were built-in features for dealing with formatting).

This looks like it's not too difficult to implement properly (and also looks like it would help fix #2), so I'll work on it as soon as I get some free time.

**Comment by saghm** [comment]
I rewrote the code to use serde_json::ser::Formatter in 02a0cf8147dfeacb459a8ef1f5e2aa608511853b, and it seems to be working the same as before (with the addition of colored escape sequences as suggested in #2). Let me know if you have any other suggestions!
